modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_showcase_search\ExistingSite;
+
+use Drupal\Core\Url;
+use Drupal\Tests\oe_showcase\ExistingSite\ShowcaseExistingSiteTestBase;
+
+/**
+ * Tests the OEL Search feature integration.
+ */
+class IntegrationTest extends ShowcaseExistingSiteTestBase {
+
+  /**
+   * Test the search page.
+   */
+  public function testSearchFeature() {
+    $assert = $this->assertSession();
+    $page = $this->getSession()->getPage();
+
+    // Assert the correct elements are in place.
+    $this->drupalGet(Url::fromRoute('<front>'));
+    $search_form = $assert->elementExists('css', '#oe-whitelabel-search-form');
+    $search_input = $search_form->findField('Search');
+    $this->assertNotNull($search_input);
+    $this->assertTrue($search_input->hasClass('form-autocomplete'));
+    $this->assertTrue($search_input->hasClass('required'));
+    $search_input->setValue('Imputo');
+    $search_button = $search_form->find('css', '#submit');
+    $this->assertNotNull($search_button);
+    $search_button->click();
+
+    $view = $assert->elementExists('css', '.views-element-container');
+    $items = $view->findAll('css', '.card-title');
+    $this->assertCount(2, $items);
+    $this->assertSame('Imputo Neo Sagaciter', $items[0]->getText());
+    $this->assertSame('Gemino Imputo', $items[1]->getText());
+
+    // @todo Use 'Clear' once OEL-1315 is resolved.
+    $this->drupalGet('/search');
+    $offcanvas = $assert->elementExists('css', '#bcl-offcanvas');
+    $title = $offcanvas->find('css', 'h4');
+    $this->assertSame('Filter Options', $title->getText());
+    $assert->fieldExists('Category 1');
+    $assert->fieldExists('Category 2');
+    $assert->fieldExists('Category 3');
+    $assert->fieldExists('Category 4');
+    $assert->fieldExists('Category 5');
```

Comment:
We can be more precise:

```suggestion
    $assert->fieldExists('Category 1', $offcanvas);
    $assert->fieldExists('Category 2', $offcanvas);
    $assert->fieldExists('Category 3', $offcanvas);
    $assert->fieldExists('Category 4', $offcanvas);
    $assert->fieldExists('Category 5', $offcanvas);
```